lmctl/project/processes/package.py

```diff
@@ -38,23 +43,35 @@ def execute(self):
         files.clean_directory(build_tree.root_path)
         pkg_meta_file_path = build_tree.pkg_meta_file_path()
         self.__create_pkg_meta(pkg_meta_file_path)
-        pkg_path = build_tree.gen_pkg_path(self.project.config.full_name, self.project.config.version)
+        pkg_path = build_tree.gen_pkg_path(self.project.config.full_name, self.project.config.version, packaging=self.project.config.packaging)
         self.journal.event('Creating package at: {0}'.format(pkg_path))
         pkg_tree = pkgs.ExpandedPkgTree()
         compiled_content_path = self.content_tree.root_path
-        with tarfile.open(pkg_path, mode="w:gz") as pkg_tar:
-            rootlen = len(compiled_content_path) + 1
-            for root, dirs, filelist in os.walk(compiled_content_path):
-                for file_name in filelist:
-                    full_path = os.path.join(root, file_name)
-                    pkg_tar.add(full_path, arcname=full_path[rootlen:])
-            pkg_tar.add(pkg_meta_file_path, arcname=pkg_tree.pkg_meta_file_name)
+        if self.project.config.packaging == CSAR_PACKAGING:
+            with zipfile.ZipFile(pkg_path, mode='w') as pkg_zip:
+                self.__build_package(pkg_zip.write, pkg_tree, compiled_content_path, pkg_meta_file_path)
+        else:
+            with tarfile.open(pkg_path, mode='w:gz') as pkg_tar:
+                self.__build_package(pkg_tar.add, pkg_tree, compiled_content_path, pkg_meta_file_path)
         self.__clear_compile_directory()
         try:
             return pkgs.Pkg(pkg_path)
         except pkgs.InvalidPackageError as e:
             raise PkgProcessError(str(e)) from e
 
+    def __build_package(self, add_method, pkg_tree, compiled_content_path, pkg_meta_file_path):
+        rootlen = len(compiled_content_path) + 1
+        for root, dirs, filelist in os.walk(compiled_content_path):
+            for file_name in filelist:
+                full_path = os.path.join(root, file_name)
+                file_size = os.path.getsize(full_path)
+                arcname = full_path[rootlen:]
+                if file_size > 100000000:
+                    # For big files let people know. TODO: make this more generic, so we can report long running tasks as events
```

Comment:
is this TODO just considered outside of the scope of this issue?

Reply:
Yeah out of scope. Just a reminder as to why we log anything. This was actually already in the code and just got moved.